src/compiler/checker.ts

```diff
@@ -15340,11 +15340,23 @@ namespace ts {
              * * Ternary.False if they are not related.
              */
             function isRelatedTo(originalSource: Type, originalTarget: Type, reportErrors = false, headMessage?: DiagnosticMessage, intersectionState = IntersectionState.None): Ternary {
+                // Before normalization: if `source` is type reference (ergo an object), and `target` is primitive,
+                // skip all the checks we don't need and just return `isSimpleTypeRelatedTo` result
+                if (getObjectFlags(originalSource) & ObjectFlags.Reference && originalTarget.flags & TypeFlags.Primitive) {

```

Comment:
`getObjectFlags` on an intersection will bubble up `ObjectFlags.Reference`. That means that a primitive branded with a type reference (maybe rare) will no longer be compatible, right?